pkg/devconsole/webhooks/webhooks.go

```diff
@@ -0,0 +1,216 @@
+package webhooks
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/openshift/console/pkg/auth"
+	"github.com/openshift/console/pkg/devconsole/common"
+)
+
+func CreateGithubWebhook(r *http.Request, user *auth.User) (common.DevConsoleCommonResponse, error) {
+	// POST /api/dev-console/webhooks/github
+	var request GithubWebhookRequest
+	err := json.NewDecoder(r.Body).Decode(&request)
+	if err != nil {
+		return common.DevConsoleCommonResponse{}, fmt.Errorf("failed to parse request: %v", err)
+	}
+
+	bodyBytes, err := json.Marshal(request.Body)
+	if err != nil {
+		return common.DevConsoleCommonResponse{}, fmt.Errorf("failed to marshal request body: %v", err)
+	}
+
+	GH_WEBHOOK_URL := fmt.Sprintf("%s/repos/%s/%s/hooks", request.HostName, request.Owner, request.RepoName)
```

Comment:
Do we need to URL encode any of these parameters?

Reply:
Not needed here.

Reply:
Let's encode the path elements anyway. It's easy to do, and it's a good policy to never trust values from the HTTP request on the backend. Then it's never a problem.